main.go

```diff
@@ -149,7 +150,7 @@ func main() {
 		Client: mgr.GetClient(),
 		Log:    ctrl.Log.WithName("controllers").WithName("infrastructure").WithName("NestedCluster"),
 		Scheme: mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
+	}).SetupWithManager(ctx, mgr, concurrency(10)); err != nil {
```

Comment:
Do you think we can make the `concurrency` configurable (through the command-line option) ?

Reply:
I agree, I think we should have this configurable. We can take the `cluster-concurrency` flag from CAPI - https://github.com/kubernetes-sigs/cluster-api/blob/master/main.go#L125-L126

Reply:
I added a todo here actually in concurreny implemention function ..
will see how we can add this